lib/http-context.js

```diff
@@ -140,24 +140,33 @@ HttpContext.prototype.buildArgs = function(method) {
 
 HttpContext.prototype.getArgByName = function(name, options) {
   var req = this.req;
-  var args = req.param('args');
+  var args = (req.params && req.params.args) ||
+    (req.body && req.body.args) ||
+    (req.query && req.query.args);
 
   if (args) {
     args = JSON.parse(args);
   }
 
-  if (typeof args !== 'object' || !args) {
+  if (typeof args !== 'object') {
     args = {};
   }
 
+  /*
   var arg = (args && args[name] !== undefined) ? args[name] :
             this.req.param(name) !== undefined ? this.req.param(name) :
             this.req.get(name);
+  */
   // search these in order by name
   // req.params
   // req.body
   // req.query
   // req.header
+  var objectsWithArg = [args, req.params, req.body, req.query].filter(function(obj) {
+    return (obj && obj[name] !== undefined);
```

Comment:
We have to be pretty conscious of performance in this function since it is used so heavily. Can you refactor this to remove the closure (`name`)?